### PR TITLE
Forbid creating clusters with more than 100 nodes without vpc-native

### PR DESCRIPTION
**Pull request by marseel**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Currently GCE cannot support large clusters without vpc-native. Therefore this PR is adding check to make sure nobody is trying to create such cluster. 
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
/sig scalability
/cc @mborsz 
/cc @wojtek-t 

```docs

```

### CODE REVIEW

File: cluster/gce/config-default.sh [hunk 1]
```diff
@@ -354,6 +354,9 @@ elif [[ -n "${MAX_PODS_PER_NODE:-}" ]]; then
   # Should not have MAX_PODS_PER_NODE set for route-based clusters.
   echo -e "${color_red:-}Cannot set MAX_PODS_PER_NODE for route-based projects for ${PROJECT}." >&2
   exit 1
+elif [[ "$(get-num-nodes)" -gt 100 ]]; then
```
Comment (wojtek-t) [inline cluster/gce/config-default.sh:357]: nit: can you change that int:
```
if [ "${ENABLE_IP_ALIASES}" = true ]; then
 ...
else
  if [[ -n "${MAX_PODS_PER_NODE:-}" ]]; then
    ..
  fi
  if [[ <your condition>]]; then
    ..
  fi
fi
```

It's easier to follow that those conditions are basically not really connected.

Comment (marseel) [reply cluster/gce/config-default.sh:357]: Makes sense, done.

**Comment by marseel** [comment]
I've only rebased this branch, because test jobs were exiting with error "merge: 560397e0600a7ef36995b6c097e8ca5d1c54850a - not something we can merge"

**Pull request merged**